stateMutators.js

```diff
@@ -0,0 +1,111 @@
+const gameState = {
+    basketPos: 50,
+    basketValue: 0,
+    score: 0,
+    livesRemaining: 3,
+    gameLevel: 1,
+    gameMode: 'running',
+    fallingObjects: [],
+}
+
+const setBasketPos = (state, value) => ({
+    ...state,
+    basketPos: value,
+});
+
+const moveBasketLeftKey = (state, value) => ({
+    ...state,
+    basketPos: basketPos - value,
+});
+
+const moveBasketRightKey = (state, value) => ({
+    ...state,
+    basketPos: basketPos + value,
+});
+
+const moveBasketMouse = (state, value) => ({
+    ...state,
+    basketPos: value,
+});
+
+const setBasketValue = (state, value) => ({
+    ...state,
+    basketValue: value,
+});
+
+const resetBasketValue = (state) => ({
+    ...state,
+    basketValue: 0,
+});
+
+const setScore = (state, value) => ({
+    ...state,
+    score: value,
+});
+
+const resetScore = (state) => ({
+    ...state,
+    score: 0,
+});
+
+const setLivesRemaining = (state, value) => ({
```

Comment:
In the context of our game I think lives only either:
- decrease, when you are penalized
- reset, when you start a game

So rather than a `setLivesRemaining` "setter" style mutator, maybe a `loseLife` mutator that always minuses 1 from `livesRemaining` state makes more sense. 

(not saying you're necessarily wrong here - this would still work fine, and could be more flexible if maybe we add more ways later to get/lose lives - but just an example of how state mutations can be thought of differently when relating back to the program's functionality)